### DISCUSSION TITLE
almost crazy trying to figure out how to get llama.cpp to prompt with mark down formatted encyclopedia style article.

### DISCUSSION THREAD

**Discussion by hiqsociety** [ggml-org/llama.cpp#3218] [Q&A]
can anyone help make llama.cpp generate markdown formatted encyclopedia style article?

./main -m models/llama-2-7b-lora-assemble.Q4_K_M.gguf -ngl 35 -c 3620 -n 12288 -p "Detailed markdown-format encyclopedia article titled 'thomas edison' with a minimum of 2400 words in English. Ensure accuracy, avoid speculation. The article is in formatted in markdown with bold, italics, heading and subheadings etc." -e -t 1

**Comment by staviq** [comment]
I'd like to point out, you are trying to give it a complicated problem to solve, and you won't be able to tell where it fails if there are multiple ways it can fail, with a complex prompt like this. And if you can't tell which parts isn't working, you can't tell which part needs fixing.

I would highly recommended you test with simple, separate prompts for each part of the problem. Casually ask it about Thomas Edison. Ask it to write a simple Wikipedia entry on something it knows for certain, like "what's a computer". Try writing separate, simple prompts asking it to write something long and detailed.

Then, also separately, play with BNF grammars. It's a great tool for making the model follow a certain format or style. But if you have newer used BNF, it does take a bit to learn how it works.

Test and play with simpler things, untill you develop an intuition for how prompts influence the LLM behaviour.

Splitting your problem into smaller ones, and lots of testing with simple prompts, makes it much much easier to create complex prompts in the end, because then, you already have working "building blocks" for more complex prompts.

**That said, to answer your actual question:**

You could try giving it BNF grammar for Wikipedia formatting, this might be helpful: https://m.mediawiki.org/wiki/Markup_spec/BNF

This should help you forcing the model to write Wikipedia formatted replies.
